segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/SegmentStateStore.java

```diff
@@ -31,6 +32,22 @@
 
 /**
  * Stores and Retrieves Segment Attribute values.
+ *
+ * Each state is represented by two files. At least one of these will always be carrying valid data.
+ *
+ * A call to put() will:
+ *  1. If more than one statefiles exist:
+ *      a. Validate that the latest one has correct data.
+ *      b. Delete the older statefile if the latest one has valid data.
+ *  2. Create non-existing statefile write state to it.
```

Comment:
"Create non-existing statefile" -> "Creates a new file with the same name as the one deleted"

Reply:
Fixed.